### PR TITLE
Make this work on python 2.7

### PR DESCRIPTION
**Pull request by nokernel**
Running on python 2.7 I was not able to import this module as it seems to require something that does not exist anymore in python 2.7.

### CODE REVIEW

**Comment by ramnes** [comment]
Hi @nokernel, thank you for this PR!

But Python 2.7 EOL is at the end of the year... I'm not sure that we want to support it. @Shir0kamii, what do you think?

If we want to support Python 2 and Python 3 at the same time, there are two problems with your merge request as is:
* `setup.py` must be updated so that `contextlib2` is added as a dependency for Python 2 installs only. 
* For Python 3 users, `contextlib` should be used.

**Comment by nokernel** [comment]
> Hi @nokernel, thank you for this PR!
> 
> But Python 2.7 EOL is at the end of the year... I'm not sure that we want to support it. @Shir0kamii, what do you think?
> 
> If we want to support Python 2 and Python 3 at the same time, there are two problems with your merge request as is:
> 
>     * `setup.py` must be updated so that `contextlib2` is added as a dependency for Python 2 installs only.
> 
>     * For Python 3 users, `contextlib` should be used.

This might be an issue for long term supported OS versions like Centos.

I understand that they could start using pyenv but hey that made it work for my python 2.7.

There is surely a better way to do this than what I did, like detecting python version and use the right lib/requirements?

This way we dont break python 2.7 even if it will be EOL ahead of it being EOL.

**Comment by ramnes** [comment]
> There is surely a better way to do this than what I did, like detecting python version and use the right lib/requirements?

Indeed, and I've given you the two things to change so that it works for both Python 2 and Python 3 users.

Although, there is another solution which I prefer that would be to not use `ExitStack` in the first place.

**Comment by nokernel** [comment]
> > There is surely a better way to do this than what I did, like detecting python version and use the right lib/requirements?
> 
> Indeed, and I've given you the two things to change so that it works for both Python 2 and Python 3 users.
> 

Yes my bad ;) I will see how to implement that.

**Comment by nokernel** [comment]
Updated to work with python 2 and 3.

Really im not very good with python, not sure this is the right way to do this.

File: setup.py [hunk 1]
```diff
@@ -1,6 +1,21 @@
...
+    'contextlib2'
+]
+
+req3 = []
+
+if major == 2:
+    req=req2
+else:
+    req=req3
+
```
Comment (ramnes) [inline setup.py:9]: Let's make something simpler like this:

```python
install_requires = []
if sys.version_info.major == 2:
    install_requires.append('contextlib2')
```

File: graphitesend/block_metric.py [hunk 2]
```diff
@@ -1,4 +1,8 @@
-from contextlib2 import contextmanager, ExitStack
+try:
+  from contextlib import contextmanager, ExitStack
+except ImportError:
+  from contextlib2 import contextmanager, ExitStack
+
```
Comment (ramnes) [inline graphitesend/block_metric.py]: Let's use `sys.version_info.major` here also and move that logic after the other imports.

Comment (nokernel) [reply graphitesend/block_metric.py]: ok let me transform that

**Comment by ramnes** [comment]
@nokernel we're almost there!

File: setup.py (hunk 1 above)
Comment (nokernel) [reply setup.py:9]: sorry I missed that comment. updating

**Comment by nokernel** [comment]
@ramnes there updated with your last suggestions.

I also fixed flake8 complaining on my changes.

But still travis if failing on some python version

**Comment by ramnes** [comment]
The Travis fails are unrelated. Thanks!

**Comment by nokernel** [comment]
@ramnes @Shir0kamii How do I get a release on pypi for this?

**Comment by ramnes** [comment]
@nokernel 0.11.2 released :wink:

**Comment by nokernel** [comment]
> @nokernel 0.11.2 released wink

Nice! thanks!

**Pull request merged**